src/backend/feed/processor.js

```diff
@@ -186,6 +188,11 @@ module.exports = async function processor(job) {
     // Version info for this feed changed, so update the database
     feed.etag = feed.etag || info.etag;
     feed.lastModified = feed.lastModified || info.lastModified;
+    // If feed.link is empty or there are blog posts
+    if (!feed.link && articles.length > 0) {
```

Comment:
How careful should we be here about articles existing, and being an array, but the first element not having both `meta` and `link`?

We could add some more protection so we don't crash here:

```js
if (!feed.link && articles.length) {
  let article = articles[0];
  let meta = article.meta;
  feed.link = meta ? meta.link : null;
}
```